exp/runtime/hooks/api/v1alpha1/lifecyclehooks_types.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	runtimecatalog "sigs.k8s.io/cluster-api/internal/runtime/catalog"
+)
+
+// BeforeClusterCreateRequest is the request of the BeforeClusterCreate hook.
+// +kubebuilder:object:root=true
+type BeforeClusterCreateRequest struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// The cluster object the lifecycle hook corresponds to.
+	Cluster clusterv1.Cluster `json:"cluster"`
+}
+
+// BeforeClusterCreateResponse is the response of BeforeClusterCreate hook.
+// +kubebuilder:object:root=true
+type BeforeClusterCreateResponse struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// CommonResponse contains a Status and Message field common to all response types.
```

Comment:
```suggestion
	// CommonResponse contains a Status and Message fields common to all response types.
```
nit

(same comment on all other uses)

Reply:
Kilians suggestion sounds like the best option to me, but I trust the native speaker on that one :)